docs/user-guide/configmap.md

```diff
@@ -0,0 +1,422 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# ConfigMap
+
```

Comment:
I think we can use a more friendly intro. For example:

Most applications require configuration data in order to operate or adjust runtime behavior. Configuration data is often exposed to applications via configuration files or retrieved from a configuration store such as etcd or consul. In both cases configuration data must be defined and made accessible to applications at the time of deployment. To keep application portable users should avoid "baking" configuration data in containers. The ConfigMap API automates the process of defining configuration data and transporting it to pods.

Something like that. I really want the user to understand why config maps exist and what problems they solve.